substrate/primitives/trie/src/lib.rs

```diff
@@ -37,6 +38,14 @@ mod trie_stream;
 #[cfg(feature = "std")]
 pub mod proof_size_extension;
 
+#[cfg(feature = "std")]
+pub use std::hash::RandomState;
+
+#[cfg(not(feature = "std"))]
+pub use hasher_random_state::RandomState;
+
+pub use hasher_random_state::add_extra_randomness;
```

Comment:
```suggestion
pub use hasher_random_state::{RandomState, add_extra_randomness};
```

This way no one accidentally uses this in `std`.